modules/app_component/variables.tf

```diff
@@ -1,3 +1,14 @@
+variable "cluster_type" {
```

Comment:
Please move down the variable declaration a bit. Look, in which sequence you're using the variable in the main.tf file (where you call this module). The order should be in sync for better readability.